### PR TITLE
David Libre: Version 1.100; ttfautohint (v1.8.4.7-5d5b) added

### LINKED ISSUE #6518: Update David Libre
**Font Project Git Repo URL:** https://github.com/meirsadan/david-libre/

**Current version:** 1.003

**New available version:** 1.100

**Super short description of the upgrade:** Fix bad positioning of diacritics under Yiddish digraphs; fix component spacing for Yiddish ligatures; change qamats to "qamats qatan" form; add dlig feature for Yiddish ligatures and yod hiriq; add final kaf with dagesh, sheva and qamats forms.

**Bugs reported in this repo which this upgrade would fix:** (in David Libre repo:)
> https://github.com/meirsadan/david-libre/issues/19
> https://github.com/meirsadan/david-libre/issues/17
> https://github.com/meirsadan/david-libre/issues/16

Thanks to font designer/developer @meirsadan for the fixes and new version.
Thanks in advance to @emmamarichal for Google Fonts help.

### PR DESCRIPTION
**Pull request by emmamarichal**
874068e: [gftools-packager] David Libre: Version 1.100; ttfautohint (v1.8.4.7-5d5b) added

* David Libre Version 1.100; ttfautohint (v1.8.4.7-5d5b) taken from the upstream repo https://github.com/meirsadan/david-libre at commit https://github.com/meirsadan/david-libre/commit/f51b8b679aeb1deac9a2f53be25487895593de58.

Closes #6518

### CODE REVIEW

File: ofl/davidlibre/OFL.txt [hunk 1]
```diff
@@ -1,8 +1,8 @@
-Copyright (c) 2014 Monotype Imaging Inc. (licensing@monotype.com). All Rights Reserved. David Hadash is a trademark of Monotype Imaging Inc. Reserved Font Name 'Hadash'. Copyright (c) 2003-2008 SIL International (http://www.sil.org/), with Reserved Font Names 'Gentium' and 'SIL'. Copyright (c) 2015-2016 Meir Sadan (meir@sadan.com).
```
Comment (RosaWagner) [inline ofl/davidlibre/OFL.txt:1]: @emmamarichal sorry I didn't notice before, the reserved font name shouldn't be removed if part of the font is indeed forked from Gentium. We should have "With reserved font name "Hadash", "Gentium" and 'SIL'." after the copyright string in OFL.txt. But not necessary to include the RFN mention in the font info.

**Comment by vv-monsalve** [comment]
> If I put 1599, we have a diff. So I kept 1528, which is the value I found in the current version in the API. So, I don't where the check found this 1599 value 🤔

The reason why it is requesting the font to have 1599 is because it was the previous Win value and the original font didn't have the Use of Type Metrics enabled. When [updating such cases](https://googlefonts.github.io/gf-guide/metrics.html#2-recalculating-the-vertical-metrics-for-an-upgraded-family) we expect the upgrade font to inherit the previous metrics values.

**Comment by vv-monsalve** [comment]
The new version is showing tofu in Hebrew (which is the main script for this font)

diffbrowsers_text-Windows-10-10 0 20348-SP0_firefox_120 0

**Comment by emmamarichal** [comment]
@vv-monsalve Ah yes, sorry I forgot to link this: https://github.com/meirsadan/david-libre/pull/22#issuecomment-1833525122

**Comment by RosaWagner** [comment]
Indeed, to summurize the doc, for the VM we would expect:
- `use typo metrics` to be enabled
- `Win` metrics values to be transferred to the `typo` metrics
- change `hhea` metrics to match the new `typo` metrics (which are the old Win metrics), because Mac user are less numerous than Windows users
- update win metrics to y-min and y-max

**Comment by emmamarichal** [comment]
@RosaWagner Yep, I know this, but when I did it, I had a visual difference. I'll check again tomorrow

**Comment by vv-monsalve** [comment]
> @vv-monsalve Ah yes, sorry I forgot to link this: [meirsadan/david-libre#22 (comment)](https://github.com/meirsadan/david-libre/pull/22#issuecomment-1833525122)

In that case, it will cause a regression, so we should check how permissive or flexible we can be in these cases.
Sadly, it seems we don't have a Hebrew glyphset defined.
@RosaWagner wdyt?

**Comment by vv-monsalve** [comment]
> VM: The difference will be expected in Mac, but diff images from Windows should show the same VM as before. Did you mean that you had a difference in Windows?

Currently yes (above image), because the font is keeping the original TypoMetrics. It will get resolved once the new font is changed to follow the spec.

On Glyphsets.

Confirming: I'm getting option 2. would apply if the answer to 1. is "yes". Is it?

**Comment by RosaWagner** [comment]
@vv-monsalve yes, option 2 would apply if 1 is yes, and if we don’t choose option 3. I don’t know what’s best between 2 and 3. I might be good to talk about it at the meeting tomorrow, but since the font was like 2 before, we may as well just continue this way (simplest solution but not the most accurate for users). It could added in the description that Yiddish is not supported.

**Comment by markhdavid** [comment]
> 1. https://github.com/googlefonts/lang/blob/main/Lib/gflanguages/data/languages/yi_Hebr.textproto

Sorry, but I'm new here. I don't understand. What is the problem? I sounds like you're considering dropping support for Yiddish. I don't understand why that would be. The original corrections to this font that prompted this attempt to push out a new version for Google Fonts was specifically to support Yiddish. Can anyone explain what is going on? I have Yiddish language and Hebrew script expertise.

**Comment by emmamarichal** [comment]
@vv-monsalve @markhdavid, I just checked the Hebrew subset, here are the glyphs not included and not supported:

ayinaltone-hb
alefWide-hb
daletWide-hb
heWide-hb
kafWide-hb
lamedWide-hb
memFinalWide-hb
reshWide-hb
tavWide-hb
yodhiriq-hb
yodyodpatah-hb
perafe-hb
kafrafe-hb
betrafe-hb
aleflamed-hb

**Comment by RosaWagner** [comment]
@markhdavid We detected multiple issues in multiple places affecting the API but I believe we need to reassure you on the nature of these issues: the way the Google Fonts API functions makes it almost impossible to drop language support, we can only add language coverage or keep the same as before. 

Now everyone bare with me, it is a bit hard to explain.

## About the cantillation marks:
What I understand from this issue https://github.com/meirsadan/david-libre/pull/22#issuecomment-1833606647, is that Yiddish has never been **fully** supported by this family because of the lack of cantillation marks. I also don't see this issue petitioning for the addition of these glyphs to be resolved: https://github.com/meirsadan/david-libre/issues/14. The cantillation marks seem to have been replaced by zero-width glyphs in the past to avoid the apparition of squares in sample texts. I believe that this is misleading users and that we should be more explicit about the lack of support for these marks. Another consideration is that we try to avoid tofu appearing in specimen pages.

1. **If you are saying that the cantillation marks are not necessary for the support of Yiddish**, I suggest to make sure that these codepoints are not used in the sample text used to represent the language: https://github.com/googlefonts/lang/blob/main/Lib/gflanguages/data/languages/yi_Hebr.textproto.
2. **If they are not used by the sample text:** then they should be removed from the font instead of having them zero-width and misleading users about the glyphset support of the font.
3. **If they are used in the sample text** then we need to either: keep the zero-width glyphs and be explicit in the description about this lack of support, or exclude this specific sample text for the specimen page (hard to do)
4. **If they are used in the sample text but not necessary for the support of the language**: we could eventually consider modifying the sample text so these codepoints don't appear.
5. **If now you mean that the cantillation marks were in fact included before**: we would need the commit where the sources were still having them so we can include them back in the last sources.

So **the first step** that you could help us with @markhdavid is to confirm if these cantillation marks are displayed in one of the sentences in the `sample_text{…}` part of [the file used to define Yiddish](https://github.com/googlefonts/lang/blob/main/Lib/gflanguages/data/languages/yi_Hebr.textproto). You are the only expert in this thread, and even if we try our best, we might miss something. Thanks a lot for your intervention.

## About the API subset definition
Apparently some Hebrew codepoints are missing from the subset definition, which is a separate issue. All non-included codepoints will be subsetted by API, meaning they would be missing from the font served by the API (not the version you can download and self-host). We need to make sure the subset is complete so the font is not savagely subsetted.

1. If the sample texts for Hebrew include some of these missing codepoints we need to rectify the subset asap, before merging this PR.
2. If the sample text for Hebrew don't include any of the missing codepoints, we can rectify that a bit later and it doesn't prevent this PR from being merged.

**Comment by markhdavid** [comment]
Hi @RosaWagner - I can confirm that cantillation marks are not used in the sample text used to represent the Yiddish language (https://github.com/googlefonts/lang/blob/main/Lib/gflanguages/data/languages/yi_Hebr.textproto). I can also assure you that cantillation marks are not necessary for the support of Yiddish.

**Comment by markhdavid** [comment]
> @vv-monsalve @markhdavid, I just checked the Hebrew subset, here are the glyphs not included and not supported:
> 
> ayinaltone-hb alefWide-hb daletWide-hb heWide-hb kafWide-hb lamedWide-hb memFinalWide-hb reshWide-hb tavWide-hb yodhiriq-hb yodyodpatah-hb perafe-hb kafrafe-hb betrafe-hb aleflamed-hb

Hi @emmamarichal.

This list contains a few characters that are precomposed variants of character combinations used for Yiddish. The full list of combinations used for Yiddish in its standard orthography, popularly known as "YIVO orthography", is shown here: http://uyip.org/combinations/ The characters in this list corresponding to the combinations used for Yiddish are the following four:
```
yodhiriq-hb yodyodpatah-hb kafrafe-hb betrafe-hb
```

All characters in this list are all in the block of Unicode called alphabetic presentation forms. That block contains, among others, characters that are precomposed versions of all the base/mark combinations required for Yiddish, but those versions are not strictly necessary to support Yiddish, since all the same combinations can be composed from base and mark characters found in Unicode's Hebrew block.

Note that none of these alphabetic-presentation-form characters are included in the `lang` repo's Yiddish language test. 

All of that said, some users and software processes do make use of precomposed characters for Yiddish for a variety of reasons, and therefore users may innocently find them included in the text of their documents that seem to contain ordinary Yiddish text, and if they happen to use a font that is missing these characters, they are typically disappointed to find that the glyph gets substituted (with that of another font, or "tofu"). Therefore, it would be nice for this and all fonts to support them. Perhaps it should be considered for the `lang` repository to add the subset of precomposed characters that correspond to composed combinations used for Yiddish to its requirements, but it's not the way it is done now, and I am not going to vociferously argue for that here in the middle of this PR and stop the show.

**Comment by emmamarichal** [comment]
I'm currently updating v-metrics and OFL, will push an update soon :)

**Comment by emmamarichal** [comment]
Updated

## David Libre: Version 1.100; ttfautohint (v1.8.4.7-5d5b) added

---
 7e0e54b: [gftools-packager] David Libre: Version 1.100; ttfautohint (v1.8.4.7-5d5b) added

* David Libre Version 1.100; ttfautohint (v1.8.4.7-5d5b) taken from the upstream repo https://github.com/meirsadan/david-libre at commit https://github.com/meirsadan/david-libre/commit/c54a7d0d0bd5756ca488821628f8e5a7b01423e7.

**Comment by emmamarichal** [comment]
@vv-monsalve @RosaWagner Updated!

**Comment by markhdavid** [comment]
Can anyone give an update where this stands @emmamarichal @RosaWagner ? Thank you

**Comment by emmamarichal** [comment]
@markhdavid No don't worry, we can ignore these fails. The font needs a last review, but it will be after holidays.
Thanks for your patience, 
cheers!

**Comment by markhdavid** [comment]
> @markhdavid No don't worry, we can ignore these fails. The font needs a last review, but it will be after holidays. Thanks for your patience, cheers!

Hi, Thank you @emmamarichal. Is there any progress, or may I know if anything is blocking this at this time? I note the [Google Fonts David Libre](https://fonts.google.com/specimen/David+Libre) page has not been updated.

**Comment by emmamarichal** [comment]
@markhdavid The review of this project is on the queue. We have a lot of projects to onboard, thank you for your patience!

**Review by vv-monsalve — CHANGES_REQUESTED** [review]

**Comment by vv-monsalve** [comment]
The inverted or open question mark `¿` has a wrong direction. It must be flipped horizontally

Screen Shot 2024-01-23 at 18 59 22

**Comment by emmamarichal** [comment]
PR upstream: https://github.com/meirsadan/david-libre/pull/25

**Comment by emmamarichal** [comment]
Updated

## David Libre: Version 1.100; ttfautohint (v1.8.4.7-5d5b) added

---
 b8b86ca: [gftools-packager] David Libre: Version 1.100; ttfautohint (v1.8.4.7-5d5b) added

* David Libre Version 1.100; ttfautohint (v1.8.4.7-5d5b) taken from the upstream repo https://github.com/meirsadan/david-libre at commit https://github.com/meirsadan/david-libre/commit/15496c1be77e35f3fa3d0df14a2c2c6f9adfd297.

**Comment by emmamarichal** [comment]
@vv-monsalve, PR updated! :)

**Comment by emmamarichal** [comment]
@felipesanches The fontbakery report generated here looks weird, do you have any clue?
The three fonts (bold, regular and medium) should appear, like here: https://github.com/google/fonts/pull/7052#issuecomment-1855542723

**Comment by felipesanches** [comment]
@m4rc1e, a new FontBakery release is available right now ([v0.12.0a1](https://pypi.org/project/fontbakery/0.12.0a1/)), containing fixes to some problems in the reports above. Please consider re-triggering the CI job for this PR so that we can double-check that all the problems were properly addressed.

**Comment by markhdavid** [comment]
Can anyone give a high-level status report? I.e., is this font `David Libre` now on its way to being served via Google Fonts here: https://fonts.google.com/specimen/David+Libre ?  If not, can anyone identify the holdup? Thank you.

**Comment by m4rc1e** [comment]
Update will be live in a couple of weeks or so.

**Comment by markhdavid** [comment]
> Update will be live in a couple of weeks or so.

Hi @m4rc1e, it's been close to 3 weeks since the last update, and I'm checking in to see if we're still on track for the change to go live. Could you provide an update on the current status or let me know if there have been any unforeseen delays? Your efforts are greatly appreciated, and I'm looking forward to the completion of this update.

**Comment by m4rc1e** [comment]
Apologies for the delay. It's graduated from our sandbox server so we only need to push it to the production server. I'm guessing this week or next. Unfortunately, I don't do these pushes myself. I'll ping you once it is live.

**Comment by markhdavid** [comment]
> Apologies for the delay. It's graduated from our sandbox server so we only need to push it to the production server. I'm guessing this week or next. Unfortunately, I don't do these pushes myself. I'll ping you once it is live.

I see it's live as of today. It looks good. Here's a screenshot of a prominent user (yiddishsongs.org), highlighting the corrected character sequence double-yod+patah on one particular page (https://yiddishsongs.org/ikh-hob-dikh-lib-vi-a-peysakhdikn-rosl/):

Yiddish Songs - David Libre - good - Screenshot 2024-03-26 at 9 41 20 PM

Thank you @m4rc1e, as well as @emmamarichal @RosaWagner @meirsadan @fontef, very much. This could be considered resolved now, at least as far as I'm concerned FWIW.

**Pull request merged**